src/haddock/libs/libplots.py

```diff
@@ -623,32 +631,133 @@ def find_best_struct(ss_file, number_of_struct=10):
+                dialog.querySelector("#dl").text = dl_name;
+                dialog.querySelector("#dl").setAttribute("download", dl_name);
+                dialog.showModal();
+                stage.loadFile(file_name).then(function (o) {
+                    o.addRepresentation("cartoon");
+                    o.autoView();
+                    stage.setParameters({backgroundColor: "white"});
+                    stage.handleResize();
+                });
+            }
+        </script>
+    """)
+    ngl_dialog = ("""
+        <body>
+            <dialog id="structureViewerDialog">
+                <div id="viewport" style="width:800px; height:600px;"></div>
+                <form>
+                    <a id="dl"
+                        style="position: absolute; top: 10px; left: 10px;">
+                    </a>
+                    <button style="position: absolute; top: 10px; right: 10px;"
+                            value="cancel"
+                            formmethod="dialog">
+                            X
+                    </button>
+                </form>
+            </dialog>
+        </body>
+    """)
+    return ngl_script + ngl_dialog
+
+
+def _add_viewers(df):
+    def _generate_download_link(file_name, dl_name):
+        html_code = "&#8595;"  # add icon
+        html_code += "&nbsp;"  # add space
+        html_code += f'<a href="{file_name}" download="{dl_name}">Download</a>'
+        return html_code
+
+    def _generate_view_link(file_name, dl_name):
+        html_code = "&#x1F441;"  # add icon
+        html_code += "&nbsp;"  # add space
+
+        # create event
+        event = f'''onClick="showStructure(\'{file_name}\', \'{dl_name}\')"'''
+        html_code += f'<a {event} style="cursor:pointer;">View</a>'  # add link
+        return html_code
+
+    def _format_cell(row):
+        file_name, dl_name = row.split(",")
+
+        # Correct path because after running analyse
+        # files are moved to analysis folder
+        correct_path = f"../{file_name}"
+
+        # Add suffix to download name
+        suffix = Path(file_name).suffix
+        dl_name = dl_name + suffix
+
+        # Generate html code
+        html_code = "<span>"
+        html_code += _generate_download_link(correct_path, dl_name)
+        html_code += "&nbsp;"  # add space
+        html_code += _generate_view_link(correct_path, dl_name)
+        html_code += "</span>"
+        return html_code
+
+    table_df = df.copy()
+    for col_name in table_df.columns[2:]:
+        str_number = col_name.split(" ")[1]
+        # Downloaded file name should include cluster rank and model (or
+        # structure) id Add zero pad number to cluster number for consistency
+        # with model (or structure) number
+        dl_names = []
+        for rn in df["Cluster Rank"]:
+            if rn == "Unclustered":
+                dl_name = f'unclustered_model{str_number}'
+            else:
+                dl_name = f'cluster{rn:02d}_model{str_number}'
```

Comment:
very minor, but why is this renumbering necessary? why not keeping the standard `cluster_N_model_M.pdb` nomenclature? I don't like seeing zeros in the file names eheh

Reply:
fixed!